builder/builder_test.go

```diff
@@ -307,3 +317,219 @@ func getAppHashFromEVM(ethState *state.StateDB, header *monomer.Header) (common.
 
 	return appHash, nil
 }
+
+func TestWithdrawalMessages(t *testing.T) {
+	pool := mempool.New(testutils.NewMemDB(t))
+	blockStore := testutils.NewLocalMemDB(t)
+	txStore := txstore.NewTxStore(testutils.NewCometMemDB(t))
+	ethstatedb := testutils.NewEthStateDB(t)
+
+	// Chain ID, app and genesis
+	var chainID monomer.ChainID
+	app := testapp.NewTest(t, chainID.String())
+	g := &genesis.Genesis{
+		ChainID:  chainID,
+		AppState: testapp.MakeGenesisAppState(t, app),
+	}
+
+	// Event bus
+	eventBus := bfttypes.NewEventBus()
+	require.NoError(t, eventBus.Start())
+	t.Cleanup(func() {
+		require.NoError(t, eventBus.Stop())
+	})
+	subChannelLen := 3 // TODO assert that this is correct
+	// +1 because we want it to be buffered even when mempool and inclusion list are empty.
+	subscription, err := eventBus.Subscribe(context.Background(), "test", &queryAll{}, subChannelLen)
+	require.NoError(t, err)
+
+	require.NoError(t, g.Commit(context.Background(), app, blockStore, ethstatedb))
+
+	l1InfoTxETH, depositTxETH, _ := testutils.GenerateEthTxs(t)
+	l1InfoTxBytes, err := l1InfoTxETH.MarshalBinary()
+	require.NoError(t, err)
+	depositTxBytes, err := depositTxETH.MarshalBinary()
+	require.NoError(t, err)
+	privKey := ed25519.GenPrivKey()
+
+	// The client context expects some of the following fields to be non-nil
+	interfaceRegistry := types.NewInterfaceRegistry()
+	marshaller := codec.NewProtoCodec(interfaceRegistry)
+	txConfig := authtx.NewTxConfig(marshaller, authtx.DefaultSignModes)
+
+	clientCtx := &client.Context{
+		TxConfig:          txConfig,
+		Codec:             marshaller,
+		InterfaceRegistry: interfaceRegistry,
+		SkipConfirm:       true,
+	}
+
+	s := signer.New(clientCtx, privKey)
+	require.NoError(t, err)
+	mockAcc := client.Account(authtypes.NewBaseAccount(s.AccountAddress(), privKey.PubKey(), 0, 0))
+	// Create a mock AccountRetriever
+	// The client context needs this for the signer
+	mockAccountRetriever := e2e.MockAccountRetriever{
+		Accounts: map[string]*client.Account{
+			s.AccountAddress().String(): &mockAcc,
+		},
+	}
+	clientCtx.AccountRetriever = mockAccountRetriever
```

Comment:
This may resolve a lot of my other comments so feel free to ignore them if you implement this. Instead of building all of this signer info, could we just use `GenerateBlock` like the builder test does to create the block transactions? We shouldn't need to use a signer since we're never actually executing the txs in the appchain, we're only testing that the block can be properly built.

Reply:
Fixed